src/test/java/XpathTest.java

```diff
@@ -0,0 +1,36 @@
+import automation.app.ButtonsPage;
+import automation.app.ElementsPage;
+import automation.app.WebTablesPage;
+import automation.base.BaseGUITest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class XpathTest extends BaseGUITest
+{
+    @Test
+    public void checkMessage() throws Throwable {
+        ElementsPage    elementsPage = new ElementsPage();
+        ButtonsPage     buttonsPage = elementsPage.goToButtonsPage();
+
+        buttonsPage.buttonClickMe.click();
+        Assert.assertEquals(buttonsPage.getMessageText(), "You have done a dynamic click");
+    }
+
+    @Test
+    public void checkTable() throws Throwable {
+        WebTablesPage webTablesPage = new WebTablesPage();
+        webTablesPage.addItem(
+                "John",
+                "Doe",
+                "30",
+                "name@example.com",
+                "5000",
+                "Accounting");
+
+        int lineNumber = 4;

```

Comment:
You do hardcode line number what is not good. Test related to order of records. You can improve it by two ways:
1. Get all emails from table and find number of line
2. Get record by email using xpath like that `//div[@role='gridcell' and text()='kierra@example.com']/..`

Reply:
Thanks. Good point. I thought about it, but I found it a bit complicated in comparison with the current solution, so I decided to choose the simpliest way.